blocks/api/serializer.js

```diff
@@ -51,24 +51,23 @@ export function getSaveContent( blockType, attributes ) {
 	}
 
 	// Adding a generic classname
-	const addAdvancedAttributes = ( element ) => {
+	const addGeneratedClassname = ( element ) => {
 		if ( ! element || ! isObject( element ) ) {
 			return element;
 		}
 
-		const extraProps = applyFilters( 'getSaveContent.extraProps', {}, blockType, attributes );
+		const props = applyFilters( 'getSaveContent.extraProps', { ...element.props }, blockType, attributes );
```

Comment:
Should we rename the filter to reflect that this aren't "extra" props, they're the source props themselves?

Raises a worry about changing filter names since there could be dependants 😬

Reply:
>You could argue that you receive the props and add extra props and the name is still valid. Happy to change if you think it's better.

Well, this could apply to most any filter, and I think the general tendency in PHP has been to name them without prefixes and more toward describing the source data. Also, the filters might remove options.